packages/app-utils/src/components/ThemeBootstrap.tsx

```diff
@@ -0,0 +1,46 @@
+import {
+  getDefaultBaseThemes,
+  ThemeContext,
+  useInitializeThemeContextValue,
+} from '@deephaven/components';
+import { useContext, useEffect } from 'react';
+import { getThemeDataFromPlugins } from '../plugins';
+import { PluginsContext } from './PluginsBootstrap';
+
+export interface ThemeBootstrapProps {
+  children: React.ReactNode;
+}
+
+export function ThemeBootstrap({ children }: ThemeBootstrapProps): JSX.Element {
+  const themeContextValue = useInitializeThemeContextValue();
+
+  // The `usePlugins` hook throws if the context value is null. Since this is
+  // the state while plugins load asynchronously, we are using `useContext`
+  // directly to avoid the exception.
+  const pluginModules = useContext(PluginsContext);
+  const { registerThemes } = themeContextValue;
+
+  useEffect(() => {
+    if (pluginModules == null) {
+      return;
+    }
+
+    registerThemes({
+      base: getDefaultBaseThemes(),
+      custom: getThemeDataFromPlugins(pluginModules),
+    });
+  }, [pluginModules, registerThemes]);
+
+  return (
+    <ThemeContext.Provider value={themeContextValue}>
```

Comment:
Instead of `registerThemes`, we could have a default `ThemeProvider` which accepts `themes` and `activeThemes` as keys or something. Then that provider does what the render here does and is a dumb provider other than that. `useTheme` hook pulls from that provider (in the components package) and here we just pass the plugin themes into the base provider.